### PR TITLE
Ed: add docker tests on gtfs fixture

### PR DESCRIPTION
**Pull request by antoine-de**
Add a docker test that load the google example of GTFS

### CODE REVIEW

File: source/ed/connectors/gtfs_parser.cpp [hunk 1]
```diff
@@ -804,8 +804,8 @@ void split_validity_pattern_over_dst(Data& data, GtfsData& gtfs_data) {
         for (const auto& utc_shit_and_periods: split_periods) {
             nm::ValidityPattern* vp = new nm::ValidityPattern(gtfs_data.production_date.begin());
 
-            for (const auto period: utc_shit_and_periods.second) {
-                for(boost::gregorian::day_iterator it(period.begin()); it < period.end(); ++it) {
+            for (const auto& period: utc_shit_and_periods.second) {
+                for (boost::gregorian::day_iterator it(period.begin()); it < period.end(); ++it) {
```
Comment (xlqian) [inline source/ed/connectors/gtfs_parser.cpp:808]: `auto it = std::begin(period); it < std::end(period); ++it`

Comment (pbougue) [inline source/ed/connectors/gtfs_parser.cpp:808]: Not possible, it's a constructor

File: source/ed/docker_tests/CMakeLists.txt [hunk 2]
```diff
@@ -5,27 +5,36 @@ SET(EITRI "${CMAKE_SOURCE_DIR}/eitri/eitri.py")
 # TODO: if more data.nav have to be created, refactor this to be more generic
 
 # == Nav generation ==
-SET(DATA_DIR_TO_LOAD ${FIXTURES_DIR}/ed/ntfs/)
+# this function call eitri to build a data.nav.lz4 file for the DATASET_NAME
```
Comment (pbougue) [inline source/ed/docker_tests/CMakeLists.txt]: minor: `calls`

File: source/ed/docker_tests/CMakeLists.txt [hunk 3]
```diff
@@ -38,8 +47,7 @@ target_link_libraries(ed_integration_test data georef types utils ${BOOST_LIBS}
 add_custom_target(run_test DEPENDS datanav_files ed_integration_test)
 add_custom_command(TARGET run_test
                    COMMAND ed_integration_test
-                   ARGS --log_format=XML --log_sink=results_ed_integration_test.xml --log_level=all --report_level=no --ntfs_file=${DATA_NAV_TO_CREATE})
-
+                   ARGS --log_format=XML --log_sink=results_ed_integration_test.xml --log_level=all --report_level=no ${TEST_CLI_PARAMS})
```
Comment (pbougue) [inline source/ed/docker_tests/CMakeLists.txt:50]: line length

Comment (antoine-de) [inline source/ed/docker_tests/CMakeLists.txt:50]: it's argument forwarded to the exe, I don't know how to tell cmake to consider it one line and I don't really want to dig into cmake :wink:

Comment (xlqian) [inline source/ed/docker_tests/CMakeLists.txt:50]: > I don't really want to dig into cmake

:trollface:

File: source/ed/connectors/gtfs_parser.cpp (hunk 1 above)
Comment (xlqian) [inline source/ed/connectors/gtfs_parser.cpp:808]: OK, my bad

File: source/ed/docker_tests/CMakeLists.txt [hunk 4]
```diff
@@ -5,27 +5,36 @@ SET(EITRI "${CMAKE_SOURCE_DIR}/eitri/eitri.py")
 # TODO: if more data.nav have to be created, refactor this to be more generic
 
 # == Nav generation ==
-SET(DATA_DIR_TO_LOAD ${FIXTURES_DIR}/ed/ntfs/)
+# this function calls eitri to build a data.nav.lz4 file for the DATASET_NAME
+macro(generate_nav DATASET_NAME)
```
Comment (TeXitoi) [inline source/ed/docker_tests/CMakeLists.txt:9]: :+1:

File: source/ed/docker_tests/ed_integration_tests.cpp [hunk 5]
```diff
@@ -131,3 +133,103 @@ BOOST_FIXTURE_TEST_CASE(fusio_test, ArgsFixture) {
...
+    BOOST_CHECK_EQUAL(fur_creek->uri, "stop_area:FUR_CREEK_RES");
+    BOOST_CHECK_EQUAL(fur_creek->name, "Furnace Creek Resort (Demo)");
+    BOOST_CHECK_CLOSE(fur_creek->coord.lat(), 36.425288, 0.1);
+    BOOST_CHECK_CLOSE(fur_creek->coord.lon(), -117.133162, 0.1);
+
+    for (auto sa: pt_data.stop_areas) {
+        BOOST_CHECK_EQUAL(sa->timezone, "America/Los_Angeles");
+    }
+
+    //stop point, and lines shoudl be equals too
```
Comment (TeXitoi) [inline source/ed/docker_tests/ed_integration_tests.cpp]: should

File: source/ed/docker_tests/ed_integration_tests.cpp [hunk 6]
```diff
@@ -131,3 +133,103 @@ BOOST_FIXTURE_TEST_CASE(fusio_test, ArgsFixture) {
...
+    BOOST_REQUIRE_EQUAL(pt_data.networks.size(), 1);
+    BOOST_CHECK_EQUAL(pt_data.networks[0]->name, "Demo Transit Authority");
+    BOOST_CHECK_EQUAL(pt_data.networks[0]->uri, "network:DTA");
+
+    BOOST_REQUIRE_EQUAL(pt_data.stop_areas.size(), 9);
+    const auto* fur_creek = pt_data.stop_areas_map.at("stop_area:FUR_CREEK_RES");
+    BOOST_CHECK_EQUAL(fur_creek->uri, "stop_area:FUR_CREEK_RES");
+    BOOST_CHECK_EQUAL(fur_creek->name, "Furnace Creek Resort (Demo)");
+    BOOST_CHECK_CLOSE(fur_creek->coord.lat(), 36.425288, 0.1);
+    BOOST_CHECK_CLOSE(fur_creek->coord.lon(), -117.133162, 0.1);
```
Comment (TeXitoi) [inline source/ed/docker_tests/ed_integration_tests.cpp]: that's quite loose but we don't really care.

**Pull request merged**